client/app/common/services/targetStateService.js

```diff
@@ -0,0 +1,27 @@
+/* Copyright (c) Trainline Limited, 2016. All rights reserved. See LICENSE.txt in the project root for license information. */
+'use strict';
+
+angular.module('EnvironmentManager.common').factory('targetStateService', function($rootScope, $q, $http) {
+
+ return {
+   changeDeploymentStatus: function(enable, service, role, environment) {
```

Comment:
not necessary to change, but usually with setters / updaters / changers first you would specify target of action, and value (enable) would come last